ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/BndErrorReporter.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ *  Copyright (c) 2023 Christoph Läubrich and others.
+ *
+ *  This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License 2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.pde.internal.core.bnd;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.pde.internal.core.builders.CompilerFlags;
+import org.eclipse.pde.internal.core.builders.ErrorReporter;
+import org.eclipse.pde.internal.core.builders.PDEMarkerFactory;
+
+import aQute.bnd.properties.IRegion;
+import aQute.bnd.properties.LineType;
+import aQute.bnd.properties.PropertiesLineReader;
+import aQute.service.reporter.Report.Location;
+
+public class BndErrorReporter extends ErrorReporter {
+
+	private BndDocument bndDocument;
+
+	public BndErrorReporter(IFile file) {
+		super(file);
+		IDocument document = createDocument(file);
+		if (document != null) {
+			bndDocument = new BndDocument(document);
+		}
+	}
+
+	@Override
+	protected void validate(IProgressMonitor monitor) {
+	}
+
+	public void addError(String err, Location location) {
+		reportProblem(err, CompilerFlags.ERROR, location);
+	}
+
+	public void addWarning(String warn, Location location) {
+		reportProblem(warn, CompilerFlags.WARNING, location);
+	}
+
+	private void reportProblem(String err, int severity, Location location) {
+		int line = -1;
+		if (location != null) {
+			if (location.line > 0) {
+				line = location.line;
+			} else if (location.header != null && bndDocument != null) {
+				try {
+					PropertiesLineReader reader = new PropertiesLineReader(bndDocument);
+					LineType type = reader.next();
+					while (type != LineType.eof) {
+						if (type == LineType.entry) {
+							String key = reader.key();
+							if (location.header.equals(key)) {
+								IRegion region = reader.region();
+								line = bndDocument.getLineOfOffset(region.getOffset()) + 1;
```

Comment:
Yes. Markers use `1` for the first line while documents use `0` for the first line.

Reply:
> Markers use 1 for the first line 

Why do they do that? 🤷🏻‍♂️

Reply:
I have no clue, the javadoc only says its 1-based not why this was chosen, I think because tools always refer to "Line 1" in error messages as well  :-)

https://github.com/eclipse-platform/eclipse.platform/blob/ba85b738b11b5e059f9d324f71d2652b3663007d/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/IMarker.java#L196-L197